tijanabekic/assignment4/Assignment 4.java

```diff
@@ -0,0 +1,101 @@
+import static org.junit.Assert.*;
+
+public class A4Q1 {
+
+    public static void findIslandEnd(boolean[][] grid, int i, int j, int row, int column) {
+        if (i < 0 || i >= row || j < 0 || j >= column) {
+            return;
+        }
+        if (grid[i][j]) {
+            //fload this cell and look for the and of the island
+            grid[i][j] = false;

```

Comment:
We weary of changing the input grid, because it means that calling findIslands twice would result in an answer of zero with the second call.